tests/ashley/test_models_lticontext.py

```diff
@@ -0,0 +1,115 @@
+        """Test utilities functions to generate django groups related to a LTIContext"""
+
+        context = LTIContextFactory()
+        # Create a django group (corresponding to the "student" role in the LTI context
+        # we just created)
+        initial_student_group = Group.objects.create(
+            name=f"cg__{context.id}__role__student"
+        )
+
+        initial_group_count = Group.objects.count()
+
+        # Ensure that the base group name generator respect a specific name pattern
+        base_group_name = f"cg__{context.id}"
+        self.assertEqual(base_group_name, context.base_group_name)
+
+        base_group = context.base_group()
+        # Ensure that the base group is created if it does not exist
+        self.assertEqual(initial_group_count + 1, Group.objects.count())
+        self.assertEqual(base_group_name, base_group.name)
+
+        # Get the django group corresponding to the "student" role.
+        role_group_student = context.role_group("student")
+        # Ensure that the previously created group is returned and no new group is created
+        self.assertEqual(initial_student_group, role_group_student)
+        self.assertEqual(initial_group_count + 1, Group.objects.count())
+
+        # Get the django group corresponding to the "instructor role
+        role_group_instructor = context.role_group("instructor")
+        # Ensure that the django group is created since it does not exist
+        self.assertEqual(initial_group_count + 2, Group.objects.count())
+        self.assertEqual(
+            f"{base_group_name}__role__instructor", role_group_instructor.name
+        )
+
+    def test_sync_user_groups(self):
+        """Test group synchronization"""
+
+        # Create 2 LTI Contexts
+        context1 = LTIContextFactory()
+        context2 = LTIContextFactory()
+
+        # Create an unrelated django group
+        unrelated_group = Group.objects.create(name="unrelated_django_group")
+
+        # Initialize a user with no group
+        user = UserFactory()
+        self.assertEqual(0, user.groups.count())
+
+        # Sync user groups in context1 with role "student"
+        context1.sync_user_groups(user, ["student"])
+        self.assertCountEqual(
+            [context1.base_group_name, f"{context1.base_group_name}__role__student"],
+            list(user.groups.values_list("name", flat=True)),
+        )
+
+        # Add the user to an unrelated django group
+        user.groups.add(unrelated_group)
+
+        # Sync user groups in context2 with multiple roles
+        context2.sync_user_groups(user, ["role1", "role2"])
+        self.assertCountEqual(
+            [
+                unrelated_group.name,
+                context1.base_group_name,
+                f"{context1.base_group_name}__role__student",
+                context2.base_group_name,
+                f"{context2.base_group_name}__role__role1",
+                f"{context2.base_group_name}__role__role2",
```

Comment:
Should we have a different character between the parameter name and its value?
group_name__role:role1 ? or | ? or...

Reply:
Ok, go for `:`